Latest build fails under NetBSD-current aarch64 and amd53

### Zig Version

0.12.0-dev.1736+2252dcc50

### Steps to Reproduce and Observed Behavior

Standard zig build procedure using self-built llvm-17, building using ninja:

```
git pull
cd build
cmake -DCMAKE_BUILD_TYPE=Release -DZIG_NO_LIB=ON -GNinja ..
ninja install
```

Observed behaviour:

```
$ ninja install
[6/9] Building C object CMakeFiles/zig2.dir/zig2.c.o
/home/xci/src/zig/build/zig2.c: In function ‘unicode_utf8ToUtf16Le__103960’:
/home/xci/src/zig/build/zig2.c:1185794: warning: assignment to ‘const uint16_t (*)[16]’ {aka ‘const short unsigned int (*)[16]’} from incompatible pointer type ‘const uint16_t *’ {
aka ‘const short unsigned int *’} [-Wincompatible-pointer-types]
1185794 |    t26 = t25.ptr;
        |
[8/9] Building stage3
FAILED: stage3/bin/zig /home/xci/src/zig/build/stage3/bin/zig
cd /home/xci/src/zig && /home/xci/src/zig/build/zig2 build --prefix /home/xci/src/zig/build/stage3 --zig-lib-dir /home/xci/src/zig/lib -Dconfig_h=/home/xci/src/zig/build/config.h -
Denable-llvm -Doptimize=ReleaseFast -Dstrip -Dno-lib -Dno-langref -Dno-autodocs -Dtarget=native -Dcpu=native -Dversion-string=0.12.0-dev.1736+2252dcc50
/home/xci/src/zig/lib/std/debug.zig:2399:49: error: no field named 'code' in union 'c.netbsd.siginfo_t'
/home/xci/src/zig/lib/std/c/netbsd.zig:1148:30: note: union declared here
ninja: build stopped: subcommand failed.
```

siginfo_t indeed does not have a field 'code': 

```
typedef union siginfo {
        char    si_pad[128];    /* Total size; for future expansion */
        struct _ksiginfo _info;
} siginfo_t;
```

### Expected Behavior

Working zig compiler in build/stage3/bin/zig

This is with the latest commit to debug.zig; I have at present working

```
❯ zig version
0.12.0-dev.1664+8ca4a5240

```
built on the 20th of November.

--- a/lib/std/Thread/Futex.zig
+++ b/lib/std/Thread/Futex.zig
@@ -306,7 +306,7 @@ const FreebsdImpl = struct {
         }
 
         const rc = os.freebsd._umtx_op(
-            @intFromPtr(&ptr.value),
+            @intFromPtr(&ptr.raw),
             @intFromEnum(os.freebsd.UMTX_OP.WAIT_UINT_PRIVATE),
             @as(c_ulong, expect),
             tm_size,
@@ -328,7 +328,7 @@ const FreebsdImpl = struct {
 
     fn wake(ptr: *const atomic.Value(u32), max_waiters: u32) void {
         const rc = os.freebsd._umtx_op(
-            @intFromPtr(&ptr.value),
+            @intFromPtr(&ptr.raw),
             @intFromEnum(os.freebsd.UMTX_OP.WAKE_PRIVATE),
             @as(c_ulong, max_waiters),
             0, // there is no timeout struct
@@ -354,7 +354,7 @@ const OpenbsdImpl = struct {
         }
 
         const rc = os.openbsd.futex(
-            @as(*const volatile u32, @ptrCast(&ptr.value)),
+            @as(*const volatile u32, @ptrCast(&ptr.raw)),
             os.openbsd.FUTEX_WAIT | os.openbsd.FUTEX_PRIVATE_FLAG,
             @as(c_int, @bitCast(expect)),
             if (timeout != null) &ts else null,
@@ -379,7 +379,7 @@ const OpenbsdImpl = struct {
 
     fn wake(ptr: *const atomic.Value(u32), max_waiters: u32) void {
         const rc = os.openbsd.futex(
-            @as(*const volatile u32, @ptrCast(&ptr.value)),
+            @as(*const volatile u32, @ptrCast(&ptr.raw)),
             os.openbsd.FUTEX_WAKE | os.openbsd.FUTEX_PRIVATE_FLAG,
             std.math.cast(c_int, max_waiters) orelse std.math.maxInt(c_int),
             null, // FUTEX_WAKE takes no timeout ptr
@@ -414,7 +414,7 @@ const DragonflyImpl = struct {
         }
 
         const value = @as(c_int, @bitCast(expect));
-        const addr = @as(*const volatile c_int, @ptrCast(&ptr.value));
+        const addr = @as(*const volatile c_int, @ptrCast(&ptr.raw));
         const rc = os.dragonfly.umtx_sleep(addr, value, timeout_us);
 
         switch (os.errno(rc)) {
@@ -443,7 +443,7 @@ const DragonflyImpl = struct {
         // https://man.dragonflybsd.org/?command=umtx&section=2
         // > umtx_wakeup() will generally return 0 unless the address is bad.
         // We are fine with the address being bad (e.g. for Semaphore.post() where Semaphore.wait() frees the Semaphore)
-        const addr = @as(*const volatile c_int, @ptrCast(&ptr.value));
+        const addr = @as(*const volatile c_int, @ptrCast(&ptr.raw));
         _ = os.dragonfly.umtx_wakeup(addr, to_wake);
     }
 };
@@ -461,7 +461,7 @@ const WasmImpl = struct {
             \\memory.atomic.wait32 0
             \\local.set %[ret]
             : [ret] "=r" (-> u32),
-            : [ptr] "r" (&ptr.value),
+            : [ptr] "r" (&ptr.raw),
               [expected] "r" (@as(i32, @bitCast(expect))),
               [timeout] "r" (to),
         );
@@ -484,7 +484,7 @@ const WasmImpl = struct {
             \\memory.atomic.notify 0
             \\local.set %[ret]
             : [ret] "=r" (-> u32),
-            : [ptr] "r" (&ptr.value),
+            : [ptr] "r" (&ptr.raw),
               [waiters] "r" (max_waiters),
         );
         _ = woken_count; // can be 0 when linker flag 'shared-memory' is not enabled

--- a/lib/std/debug.zig
+++ b/lib/std/debug.zig
@@ -2387,6 +2387,7 @@ fn handleSegfaultPosix(sig: i32, info: *const os.siginfo_t, ctx_ptr: ?*const any
         else => unreachable,
     };
 
+    const code = if (native_os == .netbsd) info.info.code else info.code;
     nosuspend switch (panic_stage) {
         0 => {
             panic_stage = 1;
@@ -2396,14 +2397,14 @@ fn handleSegfaultPosix(sig: i32, info: *const os.siginfo_t, ctx_ptr: ?*const any
                 panic_mutex.lock();
                 defer panic_mutex.unlock();
 
-                dumpSegfaultInfoPosix(sig, info.code, addr, ctx_ptr);
+                dumpSegfaultInfoPosix(sig, code, addr, ctx_ptr);
             }
 
             waitForOtherThreadToFinishPanicking();
         },
         else => {
             // panic mutex already locked
-            dumpSegfaultInfoPosix(sig, info.code, addr, ctx_ptr);
+            dumpSegfaultInfoPosix(sig, code, addr, ctx_ptr);
         },
     };
 

